matrix/lambdas/daemons/driver.py

```diff
@@ -21,15 +24,20 @@ def run(self, request_id: str, bundle_fqids: typing.List[str], format: str):
         :param bundle_fqids: List of bundle fqids to be queried on
         :param format: MatrixFormat file format of expression matrices
         """
-        self.dynamo_handler.create_state_table_entry(request_id, len(bundle_fqids))
+        num_expected_mappers = math.ceil(len(bundle_fqids) / self._bundles_per_worker)
+        self.dynamo_handler.create_state_table_entry(request_id, num_expected_mappers)
         self.dynamo_handler.create_output_table_entry(request_id)
 
-        for fqid in bundle_fqids:
-            bundle_uuid, bundle_version = fqid.split(".", 1)
+        iter_args = [iter(bundle_fqids)] * self._bundles_per_worker
```

Comment:
Could this and `zip_longest` be pulled out to a separate function to be more descriptive such as `_group_bundle_fqids(bundle_fqids, self._bundles_per_worker)`

Reply:
Yeah, I'll break it out. fwiw, I'm just copying the `grouper` function from [here](https://docs.python.org/3.6/library/itertools.html#itertools-recipes).